hadoop-tools/hadoop-fs2img/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSTreeWalk.java

```diff
@@ -71,14 +95,24 @@ private FSTreeIterator() {
     }
 
     FSTreeIterator(TreePath p) {
+      AclStatus acls = null;
+      Path remotePath = p.getFileStatus().getPath();
+      try {
+        acls = getAclStatus(fs, remotePath);
+      } catch (IOException e) {
+        LOG.warn(
+            "Got exception when trying to get remote acls for path {} : {}",
+            remotePath, e.getMessage());
+      }
       getPendingQueue().addFirst(
-          new TreePath(p.getFileStatus(), p.getParentId(), this, fs));
+          new TreePath(p.getFileStatus(), p.getParentId(), this, fs, acls));
     }
 
     FSTreeIterator(Path p) throws IOException {
       try {
         FileStatus s = fs.getFileStatus(root);
-        getPendingQueue().addFirst(new TreePath(s, -1L, this, fs));
+        AclStatus acls = getAclStatus(fs, s.getPath());
```

Comment:
This one we don't care about the IOE?

Reply:
Good catch. The scanner will not fail if the remote FS does not support ACLs. All other IOException fail the operation in all paths now. Thanks !

Reply:
@ashvina - is it possible to handle the exception the same way in both the constructors?